test/steps/context.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/PaesslerAG/jsonpath"
+	"github.com/cucumber/godog"
+)
+
+// registerContextSteps register all existing Context steps
+func registerContextSteps(s *godog.Suite, data *Data) {
+	s.Step(`^variable "([^"]*)" is an array of size (\d+)$`, data.variableIsArrayOfSize)
+	s.Step(`^parse variable "([^"]*)" as a map and set key which value matches "([^"]*)" into variable "([^"]*)"$`, data.parseVariableAsMapAndGetKeyFromValue)
+	s.Step(`^parse variable "([^"]*)" with expression "([^"]*)" into variable "([^"]*)"$`, data.parseVariableWithExpression)
+}
+
+func (data *Data) variableIsArrayOfSize(contextKey string, size int) error {
+	contextValue := data.ScenarioContext[contextKey]
+	var array []interface{}
+	if err := json.NewDecoder(strings.NewReader(contextValue)).Decode(&array); err != nil {
```

Comment:
I think this could be moved to separate function which will retrieve context structure based on the pointer supplied as a parameter.

Reply:
Good idea! Will do.